### ISSUE TITLE
in_statement_intentation false positive on bracket?

### ISSUE THREAD

**Issue by sbcgua** [abaplint/abaplint#3929]
<img width="1091" height="174" alt="Image" src="https://github.com/user-attachments/assets/c98dffcf-fd92-48c9-909e-8ffb9c2a6e97" />

**Comment by sbcgua** [comment]
One more (though this is for discussion - I don't know what is the best style here - quite probably there is none ... - if needed I'll separate in another issue)

```abap
    loop at ct_dataset assigning <d>
      where vbeln is initial.
      ^^^^^
      read table lt_groups into ls_group
        with key order_no = <d>-order_no binary search.
```
It resolves if 2 more spaces added, but I wouldn;t say it is more readable.

```abap
    loop at ct_dataset assigning <d>
        where vbeln is initial.
      read table lt_groups into ls_group
        with key order_no = <d>-order_no binary search.
```
This case is really arguable and there is no good solution imho, But then it would be nice to have a config option. Like "statementNextLine" or something like that.

**Comment by larshp** [comment]
https://rules.abaplint.org/in_statement_indentation/

> Lines following the first line should be indented once (2 spaces).
> 
> For block declaration statements, lines after the first should be indented an additional time (default: +2 spaces)
> to distinguish them better from code within the block.

It doesn't differentiate what parts of the statement is to be indented, it always suggest all parts to have at least the same indentation

**Comment by sbcgua** [comment]
Yeah, but the chained statement is kind of next statement. At from human perspective it is so.

The good:

```abap
callXXX(
  param1 = 123
  param2 = 123
)->next(
  param1 = 123
  param2 = 123
...
```

and this is less readable (it does not visualy separate chainlets)
```abap
callXXX(
  param1 = 123
  param2 = 123
  )->next(
  param1 = 123
  param2 = 123
...
``

**Comment by larshp** [comment]
yea, its arguable, might need another indentation rule

```abap 
XXXXXXX
  XXXXX
XXXXXXX
  XXXXX
XXXXXXX
    XXXXX
  XXXX
XXX
```

^above I'd count as 5 statements, with no ambiguity, two spaces indentation in the statement or via nesting, and 4 spaces if its a nesting statement

if not having extra spaces for statements introducing indentation, or having parts of one statement start at the same column, the above example becomes ambigious

**Comment by larshp** [comment]
@sbcgua if you write down the rules you'd like in an indentation rule, then I can prompt the LLM to add it as another rule

**Comment by sbcgua** [comment]
Hmm, not sure I understood the example. But surely the indentation is opinionated anyway :)

> if you write down the rules you'd like in an indentation rule, then I can prompt the LLM to add it as another rule

Sounds interesting, do you mean a custom rule or a flag in the `in_statement_intentation` ?

**Comment by larshp** [comment]
we can just add another, all rules are somewhat opinionated, and I dont mind if they are contradictory
